pylinalg/obj/matrix.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+
+from .base import LinalgBase
+from ..func import (
+    matrix_compose,
+    matrix_decompose,
+    matrix_inverse,
+)
+
+
+class Matrix(LinalgBase):
+    """A representation of a transformation matrix in 4D
+    homogeneous (to 3D Euclidean) space."""
+
+    def __init__(self, matrix=None, /, *, dtype=None):
+        if matrix is not None:
+            if dtype is None:
+                self._val = np.asarray(matrix)
+            else:
+                self._val = np.asarray(matrix, dtype=dtype)
+        else:
+            if dtype is None:
+                dtype = "f8"
+            self._val = np.identity(4, dtype=dtype)
+
+    def copy(self):
+        return self.__class__(self._val.copy())
+
+    def icompose(self, translation, rotation, scaling):
```

Comment:
"i" for internal, right?  On that note, I guess all these methods need a docstring :)

Reply:
> Inplace

Ah, yes that's what I meant 😊